charts/templates/deployment.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "link-shortener.fullname" . }}
+  namespace: link-shortener{{ .Values.django.env.run_env }}
+  labels:
+    {{- include "link-shortener.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "link-shortener.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "link-shortener.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          envFrom:
+            - configMapRef:
+                name: {{ include "link-shortener.fullname" . }}-configmap
+            - secretRef:
+                name: {{ include "link-shortener.fullname" . }}-secrets
+          ports:
+            - name: http
+              containerPort: 8000
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /report
+              port: http
+              httpHeaders:
+                - name: Host
+                  value: {{ range .Values.django.env.allowed_hosts }}{{ . }}{{ end }}
```

Comment:
Is this needed to get the probes working? Just curious on how this work and what the use cases are.

Reply:
@cachemeoutside It's a Django thing: https://docs.djangoproject.com/en/4.2/ref/settings/#allowed-hosts

> A list of strings representing the host/domain names that this Django site can serve. This is a security measure to prevent [HTTP Host header attacks](https://docs.djangoproject.com/en/4.2/topics/security/#host-headers-virtual-hosting), which are possible even under many seemingly-safe web server configurations.

We set it to the specific assigned domain for each of our Django applications.  It's not strictly necessary given our development processes and production infrastructure, but it's recommended and easy enough to set.